Sources/XMLCoder/Auxiliaries/XMLHeader.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  XMLHeader.swift
+//  XMLCoder
+//
+//  Created by Vincent Esche on 12/18/18.
+//
+
+import Foundation
+
+public struct XMLHeader {
+    /// the XML standard that the produced document conforms to.
+    public let version: Double?
+    /// the encoding standard used to represent the characters in the produced document.
+    public let encoding: String?
+    /// indicates whether a document relies on information from an external source.
+    public let standalone: String?
+    
+    public init(version: Double? = nil, encoding: String? = nil, standalone: String? = nil) {
+        self.version = version
+        self.encoding = encoding
+        self.standalone = standalone
+    }
+    
+    internal func isEmpty() -> Bool {
```

Comment:
another use of `internal` here

Reply:
There are 94 uses of `internal`. I removed all that were added by this PR.

Reply:
awesome, that's great, thank you!

Reply:
Wait, what I meant by that is "there still are 94 occurrences of `internal` after removing (what should be) all I added". 😅

Removing 94 occurrences of `internal` sprinkled all over `XMLCoder` adds a lot of noise to this already rather large PR.
I'd suggest doing a cleanup PR afterwards that does just that. What do you think?

Reply:
all good, thanks